### PR TITLE
SHALLOT-15 Throw clearer errors from mixin getter functions

### PR DESCRIPTION
**Pull request by Aa-thomas**
Adds clearer console warnings to getRadius, getLetterSpacing, getLineHeight, getFontSize, getUnits, getElevation

For example:
`Radius not found for value "${value}". Are you sure it's defined in your theme and you're using a ThemeProvider?`

### CODE REVIEW

File: platforms/web/styles/index.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ export const GlobalStyle = createGlobalStyle<GlobalStyleProps>`
 
   body {
     color: ${({ theme, textColor }) =>
-      getColorShade(textColor ?? 'Shading.950', theme)};
+      getColorShade(textColor ?? 'Shading.900', theme)};
```
Comment (mitchellbutler) [inline platforms/web/styles/index.ts:59]: Was this change intentional? I think it's unrelated to the PR.

**Review by mitchellbutler — CHANGES_REQUESTED** [review]
Looking good! Can we make sure all of the other mixins in this file are handled too though?
It seems like `getColor` and `getColorShade` still have the issue we're trying to resolve here.

File: platforms/web/styles/index.ts (hunk 1 above)
Comment (Aa-thomas) [reply platforms/web/styles/index.ts]: Not sure tbh, but I'll change it back.

File: packages/core/src/utils/mixins/index.ts [hunk 2]
```diff
@@ -7,7 +7,9 @@ export const getColor =
     const color =
       theme.colors[value][shade as keyof typeof theme.colors.Shading]
```
Comment (mitchellbutler) [inline packages/core/src/utils/mixins/index.ts:8]: I think we need to use optional chaining here. If the component isn't wrapped in a ThemeProvider, then `theme.colors[value][shade as keyof typeof theme.colors.Shading]` will throw an error because `theme.colors` won't exist.

We should do something like:
```ts
const color = theme.colors?.[value]?.[shade as keyof typeof theme.colors.Shading]
```

**Review by mitchellbutler — CHANGES_REQUESTED** [review]

**Review by mitchellbutler — APPROVED** [review]
